stdlib/LinearAlgebra/src/cholesky.jl

```diff
@@ -153,12 +181,15 @@ end
 function _chol!(A::AbstractMatrix, ::Type{UpperTriangular})
     require_one_based_indexing(A)
     n = checksquare(A)
+    realdiag = eltype(A) <: Complex
     @inbounds begin
         for k = 1:n
+            Akk = realdiag ? real(A[k,k]) : A[k,k]
             for i = 1:k - 1
-                A[k,k] -= A[i,k]'A[i,k]
+                Akk -= realdiag ? abs2(A[i,k]) : A[i,k]'A[i,k]
             end
-            Akk, info = _chol!(A[k,k], UpperTriangular)
+            A[k,k] = Akk
+            Akk, info = _chol!(Akk, UpperTriangular)
```

Comment:
Shouldn't these two lines be swapped to ensure that it is the same as the old code, which mutates `A[k,k]` to the *final* value of `Akk`?